app/controllers/slides_controller.rb

```diff
@@ -171,22 +172,12 @@ def downloadable?
       redirect_to slide_path(@slide.id) if @slide.downloadable != true
     end
 
-    def duplicate_key_check!
-      key = params[:slide][:key]
-      if Slide.where('slides.key = ?', key).count > 0
-        redirect_to slides_path
-      end
+    def duplicate_key?
+      redirect_to slides_path if Slide.key_exist?(params[:slide][:key])
     end
 
     def slide_position
-      if params.key?(:page)
-        position = params[:page].to_i
-        if position <= 0
-          position = 1
-        end
-      else
-        position = 1
-      end
-      position
+      position = 1
```

Comment:
__[rubocop]__
* Lint/UselessAssignment: Useless assignment to variable - `position`. [:ref](https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)

<!-- Comment by [SideCI](https://sideci.com) -->